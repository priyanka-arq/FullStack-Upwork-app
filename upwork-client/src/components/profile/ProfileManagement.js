import DisplayProfile from "./DisplayProfile";
import { connect } from "react-redux";
import * as profileActions from "../../redux/actions/profileActions";
import React, { Component } from "react";
import AuthService from "../../services/auth.service";
import ProfileForm from "./ProfileForm";
import profileImage from "../Images/profileImage.png";
import styled from "styled-components";

const ProfileContainer = styled.div`
  width: 50;
  height: 50;
  justify-content: center;
`;

export class ProfileManagement extends Component {
  constructor(props) {
    super(props);
    this.state = {
      profile: {},
      redirect: "",
      user_id: this.props.match.params.userId,
      userProfile: true,
    };
  }

  componentDidMount() {
    const profiles = this.props.profiles;

    if (profiles.length === 0) {
      this.props.loadProfiles().catch((error) => {
        alert("Failed to load Profile " + error);
      });
    }

    const profile = this.props.profile;
    this.setState({ profile: profile });
  }
  handleUpdate = (event) => {
    this.props.history.push(`/profile/${this.state.profile._id}/update`);
    // if (this.props.match.params.userId) {
    //   this.props.history.push(`/profile/${this.state.user_id}/update`);
    // } else {
    //   this.props.history.push(`/profile/${this.state.profile._id}/update`);
    // }
  };

  handleClick = () => {
    this.props.history.push(`/profile`);
  };

  render() {
    return (
      <div>
        {this.state.profile ? (
          <DisplayProfile
            profile={this.state.profile}
            handleUpdate={this.handleUpdate}
          />
        ) : (
          <div>
            <h2>Would you like to create a Profile? </h2>
            <button onClick={this.handleClick}>Create Profile</button>
            <ProfileContainer>
              <img id="phoneImage" src={profileImage} />
            </ProfileContainer>
          </div>
        )}
      </div>
    );
  }
}
const mapStateToProps = (state, ownProps) => ({
  profiles: state.profiles,
  profile: state.profiles.find(
    (profile) => profile.user === ownProps.match.params.userId
  ),
});

const mapDispatchToProps = (dispatch) => ({
  loadProfiles: () => dispatch(profileActions.loadProfiles()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ProfileManagement);
