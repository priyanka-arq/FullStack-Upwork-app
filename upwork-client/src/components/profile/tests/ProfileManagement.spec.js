import React from "react";
import { shallow } from "enzyme";
import profiles from "../../../fixtures/profiles";

//import unconnected vesion of ProfileManagement and provide props
import { ProfileManagement } from "../ProfileManagement";
import toJSON from "enzyme-to-json";
//import DisplayProfile from "../DisplayProfile";
let wrapped, getCurrentProfile, loadProjects;

beforeEach(() => {
  loadProjects = jest.fn();

  wrapped = shallow(
    <ProfileManagement
      profiles={profiles}
      profile={profiles[0]}
      match={{ params: { userId: 1 } }}
      loadProjects={loadProjects}
    />
  );
});

test("should render ProfileManagement correctly", () => {
  expect(toJSON(wrapped)).toMatchSnapshot();
});

it("should render DisplayProfile", () => {
  expect(wrapped.find("DisplayProfile").length).toEqual(1);
});
