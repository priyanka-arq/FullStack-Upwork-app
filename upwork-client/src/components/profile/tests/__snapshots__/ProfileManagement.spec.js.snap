// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProfileManagement correctly 1`] = `
<div>
  <DisplayProfile
    handleUpdate={[Function]}
    profile={
      Object {
        "aboutme": "I have a great passion about solving real world problems and helping people. And now, with the skills that I have been learning in programming and computer science from the Software Engineering Course at General Assembly, I will be able to do that with new technologies. I believe in the power of technology and its impact to everyone's daily lives.",
        "email": "test@ga.com",
        "github": "https://github.com/little-chi-mai",
        "id": 1,
        "imageUrl": null,
        "linkedIn": "https://linkedin.com/in/chi-mai-nguyenn",
        "name": "Test",
        "phone": "04444444444",
        "portfolio": "https://little-chi-mai.github.io/portfolio",
        "postcode": 2000,
        "previewSource": "",
        "resume": "",
        "skills": Array [
          "React",
          "Node",
          "Java",
          "CSS",
        ],
        "suburb": "Sydney",
        "title": "Junior Developer",
        "user": 1,
      }
    }
  />
</div>
`;
